feature_tests/src/attrs.rs

```diff
@@ -312,6 +312,9 @@ pub mod ffi {
         }
     }
 
+    #[deprecated(note = "use Foo")]
+    pub struct Deprecated;
```

Comment:
suggestion: perhaps have a method instead so that it doesn't generate a new type? And you can deprecate one of the existing types if you want to test this on types too.